### PR TITLE
Improve install-cni.sh

### PR DESCRIPTION
**Pull request by jingyuanliang**

### CODE REVIEW

**Comment by jingyuanliang** [comment]
Sample log output:
```
[...]
Creating CNI spec...
CNI spec at /host/etc/cni/net.d/10-gke-ptp.conflist (ewogICJuYW1lIjogImdrZS1wb2QtbmV0d29yayIsCiAgImNuaVZlcnNpb24iOiAiMC4zLjEiLAogICJwbHVnaW5zIjogWwogICAgewogICAgICAidHlwZSI6ICJna2UiLAogICAgICAibXR1IjogMTQ2MCwKICAgICAgImlwYW0iOiB7CiAgICAgICAgICAidHlwZSI6ICJob3N0LWxvY2FsIiwKICAgICAgICAgICJyYW5nZXMiOiBbCiAgICAgICAgICBbeyJzdWJuZXQiOiAiMTAuOC4zLjAvMjQifV0KICAgICAgICAgIF0sCiAgICAgICAgICAicm91dGVzIjogWwogICAgICAgICAgICB7ImRzdCI6ICIwLjAuMC4wLzAifQogICAgICAgICAgXQogICAgICB9CiAgICB9LAogICAgewogICAgICAidHlwZSI6ICJwb3J0bWFwIiwKICAgICAgImNhcGFiaWxpdGllcyI6IHsKICAgICAgICAicG9ydE1hcHBpbmdzIjogdHJ1ZQogICAgICB9CiAgICB9LCB7InR5cGUiOiAiYmFuZHdpZHRoIiwgImNhcGFiaWxpdGllcyI6IHsiYmFuZHdpZHRoIjogdHJ1ZX19CiAgXQp9Cg==): {"name":"gke-pod-network","cniVersion":"0.3.1","plugins":[{"type":"gke","mtu":1460,"ipam":{"type":"host-local","ranges":[[{"subnet":"10.8.3.0/24"}]],"routes":[{"dst":"0.0.0.0/0"}]}},{"type":"portmap","capabilities":{"portMappings":true}},{"type":"bandwidth","capabilities":{"bandwidth":true}}]}
```
/hold cancel

**Comment by jingyuanliang** [comment]
Let's have this:
```
[...]
Creating CNI spec...
CNI spec at /host/etc/cni/net.d/10-gke-ptp.conflist, compact: {"name":"gke-pod-network","cniVersion":"0.3.1","plugins":[{"type":"gke","mtu":1460,"ipam":{"type":"host-local","ranges":[[{"subnet":"10.8.1.0/24"}]],"routes":[{"dst":"0.0.0.0/0"}]}},{"type":"portmap","capabilities":{"portMappings":true}},{"type":"bandwidth","capabilities":{"bandwidth":true}}]}
CNI spec at /host/etc/cni/net.d/10-gke-ptp.conflist, base64: ewogICJuYW1lIjogImdrZS1wb2QtbmV0d29yayIsCiAgImNuaVZlcnNpb24iOiAiMC4zLjEiLAogICJwbHVnaW5zIjogWwogICAgewogICAgICAidHlwZSI6ICJna2UiLAogICAgICAibXR1IjogMTQ2MCwKICAgICAgImlwYW0iOiB7CiAgICAgICAgICAidHlwZSI6ICJob3N0LWxvY2FsIiwKICAgICAgICAgICJyYW5nZXMiOiBbCiAgICAgICAgICBbeyJzdWJuZXQiOiAiMTAuOC4xLjAvMjQifV0KICAgICAgICAgIF0sCiAgICAgICAgICAicm91dGVzIjogWwogICAgICAgICAgICB7ImRzdCI6ICIwLjAuMC4wLzAifQogICAgICAgICAgXQogICAgICB9CiAgICB9LAogICAgewogICAgICAidHlwZSI6ICJwb3J0bWFwIiwKICAgICAgImNhcGFiaWxpdGllcyI6IHsKICAgICAgICAicG9ydE1hcHBpbmdzIjogdHJ1ZQogICAgICB9CiAgICB9LCB7InR5cGUiOiAiYmFuZHdpZHRoIiwgImNhcGFiaWxpdGllcyI6IHsiYmFuZHdpZHRoIjogdHJ1ZX19CiAgXQp9Cg==
```

File: scripts/install-cni.sh [hunk 1]
```diff
@@ -220,37 +203,36 @@ if [ "${ENABLE_PRIVATE_IPV6_ACCESS:-}" == "true" ] || [ "$ENABLE_IPV6" == "true"
...
+# Note that Destination and Mask are IP represented in hexadecimal in
+# little-endian. Regardless of endianness, 00000000 is always 0.0.0.0.
 
 # The first grep extracts the default line, and the second grep extracts the
-# last field, which is the interface name. We stick to using grep to avoid
+# first field, which is the interface name. We stick to using grep to avoid
 # introducing too many new dependencies.
 
-default_nic=$(route -n | grep -E '^0\.0\.0\.0\s+\S+\s+0\.0\.0\.0' | grep -oE '\S+$')
+default_nic=$(grep -E '^\S+\s+00000000\s+\S+\s+\S+\s+\S+\s+\S+\s+\S+\s+00000000\s+' /proc/net/route | grep -oE '^\S+')
```
Comment (aojea) [inline scripts/install-cni.sh:227]: this is not very human friendly 🤖  😄

**Comment by aojea** [comment]
my personal thoughts and experience is that this will require a much higher level of understanding of bashism and the net benefit is not much, but I defer to maintainers to decide

File: scripts/install-cni.sh (hunk 1 above)
Comment (jingyuanliang) [reply scripts/install-cni.sh:227]: The original is not much better...

**Comment by jingyuanliang** [comment]
I tried to make the change because `route` was declared "pending" in toybox although it was pending mostly because of add/del. Now it looks "completed" per https://landley.net/toybox/status.html and I'm more comfortable using it. Let me bump the toybox version first...

/hold

**Comment by jingyuanliang** [comment]
Also - https://github.com/landley/toybox/commit/c62d8a9f7471ebbb95d0bc2a6ccd6ee195719b44#commitcomment-96115890

File: scripts/install-cni.sh (hunk 1 above)
Comment (jingyuanliang) [reply scripts/install-cni.sh]: Took this part out of the PR.

**Review by MrHohn — APPROVED** [review]
Thanks Jingyuan. This is cleaner and one less dependency on sed.

Printing out CNI config also seems like a good idea.
/lgtm
/approve

**Pull request merged**